### ISSUE TITLE
Cannot install via pip on docker image python:3.10.0-slim

### ISSUE THREAD

**Issue by lippertto** [gevent/gevent#1838]
* gevent version: 
                  trying to install via pip (version 21.8.0)
* Python version:
                  Python 3.10 in the docker image python:3.10.0-slim
* Operating System: 
                  Docker image python:3.10.0-slim (run in a Ubuntu 20.10 but that should hopefully not matter so much)
The tag that I used was 1aa28c1073244b1211fcf223748c18b451890b5b95edcc2e441857ee10a8d583

### Description:

I am trying to install gevent in the python:3.10.0-slim docker image. However, the build fails, and there is no error message that I can understand.

I get the following error message 

```
 running build_ext                                                                                                                                                                                                
  generating cffi module 'build/temp.linux-x86_64-3.10/gevent.libuv._corecffi.c'                                                                                                                                   
  creating build/temp.linux-x86_64-3.10                                                                                                                                                                            
  Running '(cd  "/tmp/pip-install-nf44th2u/gevent_fe2680b4c99e4ea5a282357c116b855e/deps/libev"  && sh ./configure -C > configure-output.txt )' in /tmp/pip-install-nf44th2u/gevent_fe2680b4c99e4ea5a282357c116b855e
  config.status: error: in `/tmp/pip-install-nf44th2u/gevent_fe2680b4c99e4ea5a282357c116b855e/deps/libev':                                                                                                         
  config.status: error: Something went wrong bootstrapping makefile fragments                                                                                                                                      
      for automatic dependency tracking.  Try re-running configure with the                                                                                                                                        
      '--disable-dependency-tracking' option to at least be able to build                                                                                                                                          
      the package (albeit without support for automatic dependency tracking).                                                                                                                                      
  See `config.log' for more details                                                                                                                                                                                
<snip>
  subprocess.CalledProcessError: Command '(cd  "/tmp/pip-install-nf44th2u/gevent_fe2680b4c99e4ea5a282357c116b855e/deps/libev"  && sh ./configure -C > configure-output.txt )' returned non-zero exit status 1.
    ----------------------------------------
      ERROR: Failed building wheel for gevent

```

I assume that there is something wrong in the libev version that gevent uses here. But I think it is in the interest of the project that it gevent can be installed on official python containers.

When I build the package from source with in a running container, this works. Also, the regular image without the `-slim` works too.

If you can point me in any direction, I am happy to find out more about this problem.
         
### What I've run:

I created a Dockerfile like so:
```Dockerfile
FROM python:3.10.0-slim

RUN apt-get update && apt-get install --yes \
	gcc \
	file

RUN mkdir /install
WORKDIR /install
RUN pip install --upgrade pip
RUN pip install --upgrade setuptools
RUN pip install --prefix=/install gevent=21.8.0
```

Then I ran `docker build .`

**Comment by jamadden** [comment]
Most likely you're still missing required build-time dependencies. Without looking at the log, I can't tell you what they are, but it should be pretty obvious from the error messages the log contains. Some of the known dependencies are listed [here](https://www.gevent.org/development/installing_from_source.html).

I hope to be able to release the next minor version of gevent this month and it should include binary manylinux wheels for that platform. You can also always build your own binary wheel in a regular container and install it in a stripped container.

**Comment by lippertto** [comment]
In the end it turned out to be quite trivial. It was just hard to get to the build logs because I have no experience with setuptools. It looks like `make` is not available in the 3.10-slim image. (actually it's not on the other slim images either, I made an error  comparing the 3.10 to 3.9 ones)

Anyhow. If anyone is using gevent on a 3.10-slim image, here are the correct build dependencies:

```Dockerfile
FROM python:3.10.0-slim

RUN apt-get update && apt-get install --yes \
	gcc \
	file \
	make

RUN mkdir /install
WORKDIR /install
RUN pip install --upgrade pip
RUN pip install --upgrade setuptools
RUN pip install --prefix=/install gevent==21.8.0
```
